src/lib.rs

```diff
@@ -217,48 +215,21 @@ impl<T: Translation> Mapping<T> {
                 }
             }
         }
-
+        self.mark_active();
         previous_ttbr
     }
 
-    /// Activates the page table by setting the appropriate `TTBRn_ELx` to point to it, and saves
-    /// the previous value of `TTBRn_ELx` so that it may later be restored by
-    /// [`deactivate`](Self::deactivate).
-    ///
-    /// Panics if a previous value of `TTBRn_ELx` is already saved and not yet used by a call to
-    /// `deactivate`.
-    ///
-    /// In test builds or builds that do not target aarch64, the `TTBRn_ELx` access is omitted.
-    ///
-    /// # Safety
-    ///
-    /// The caller must ensure that the page table doesn't unmap any memory which the program is
-    /// using, or introduce aliases which break Rust's aliasing rules. The page table must not be
-    /// dropped as long as its mappings are required, as it will automatically be deactivated when
-    /// it is dropped.
-    pub unsafe fn activate(&mut self) {
-        assert!(!self.active());
-
-        // SAFETY: Our caller promises they have everything necessary mapped appropriately, and we
-        // call `mark_active` immediately after this.
-        let previous_ttbr = unsafe { self.activate_raw() };
-        self.mark_active(previous_ttbr);
-    }
-
-    /// Deactivates the page table, by setting `TTBRn_ELx` back to the value it had before
-    /// [`activate`](Self::activate) was called, and invalidating the TLB for this page table's
-    /// configured ASID.
-    ///
-    /// Panics if there is no saved `TTBRn_ELx` value because `activate` has not previously been
-    /// called.
+    /// Deactivates the page table, by setting `TTBRn_ELx` to the provided value, and invalidating
+    /// the TLB for this page table's configured ASID. The provided TTBR value should be the value
+    /// returned by the preceding [`activate`](Self::activate) call.
     ///
     /// In test builds or builds that do not target aarch64, the `TTBRn_ELx` access is omitted.
     ///
     /// # Safety
     ///
     /// The caller must ensure that the previous page table which this is switching back to doesn't
     /// unmap any memory which the program is using.
-    pub unsafe fn deactivate(&mut self) {
+    pub unsafe fn deactivate(&mut self, _previous_ttbr: usize) {
```

Comment:
Calling the parameter `_previous_ttbr` is misleading, as it is used in normal configurations. I guess you did this to avoid a warning in tests; could you instead add an appropriate `#[allow(unused)]` attribute?

Reply:
Sure. How do I attach that to a function parameter?